File: Python3/Medium/WrongAnswer/longest-palindrome-by-concatenating-two-letter-words_3.py
class Solution:
    def longestPalindrome(self, words: List[str]) -> int:
        
        d, res = collections.defaultdict(int), 0
        for w in words:
            d[w] += 1
        for w in d:
            if d[w] > 1:
                if w[0] == w[1]:
                    res += d[w] // 2 * 2
                else:
                    if d[w[::-1]] > 0:
                        res += min(d[w], d[w[::-1]]) * 2
                        d[w] = d[w[::-1]] = 0
        for w in d:
            if d[w] > 0 and w[0] == w[1]:
                res += 1
                break
        return res

File: C++/Medium/WrongAnswer/longest-palindrome-by-concatenating-two-letter-words_3.cpp
class Solution {
public:
    int longestPalindrome(vector<string>& words) {
        
        vector<int> mask(words.size(), 0);
        for(int i = 0; i < words.size(); i++) {
            for(int j = 0; j < 2; j++) {
                mask[i] |= 1 << (words[i][j] - 'a');
            }
        }
        unordered_map<int, int> counts;
        for(int i = 0; i < words.size(); i++) {
            counts[mask[i]]++;
        }
        int res = 0;
        for(auto it = counts.begin(); it != counts.end(); it++) {
            for(auto it2 = counts.begin(); it2 != counts.end(); it2++) {
                if((it->first & it2->first) == 0) {
                    res = max(res, it->second * it2->second);
                }
            }
        }
        return res;
    }
};